### ISSUE TITLE
Averaging speed when running in parallel

### ISSUE THREAD

**Issue by lkwagner** [WagnerGroup/pyqmc#274]
I noticed with @kkleiner1  that for small systems and large numbers of parameters to optimize, the code spends a lot of its time averaging or collecting data. Might be worth some further investigation.

**Comment by lkwagner** [comment]
For example, he was working on the CO molecule with ~2000 parameters to optimize, 20 cores, and 400 walkers. This makes some sense, but I would be interested to know if we can speed that part up.

**Comment by lkwagner** [comment]
Not a priority.
